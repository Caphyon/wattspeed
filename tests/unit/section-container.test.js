import { shallowMount } from "@vue/test-utils";
import section_container from "../../src/components/sections/section-container"; // name of your Vue component

describe("Testing Section-Container component", () => {
  it("Basic main container", () => {
    const wrapper = shallowMount(section_container, {
      computed: {
        title() {
          return "Title Test";
        },
        noDataMsg() {
          return "Warning Test";
        },
        error() {
          return "Error Test";
        },
        loading() {
          return true;
        },
        hasData() {
          return true;
        }
      }
    });
    // controll if the component is instance
    expect(wrapper.exists()).toBeTruthy();

    // check if the divs are presents
    expect(wrapper.find(".sections__container").exists()).toBe(true);

    // Header
    expect(wrapper.find(".sections__container--title").exists()).toBe(true);
    expect(wrapper.find(".sections__container--title > svg").exists()).toBe(
      true
    );

    // Main content
    expect(wrapper.find(".sections__container--content").exists()).toBe(true);
    expect(wrapper.element).toMatchSnapshot();
  });
  it("Loading true", () => {
    const wrapper = shallowMount(section_container, {
      computed: {
        title() {
          return "Title Test";
        },
        noDataMsg() {
          return "Warning Test";
        },
        error() {
          return "Error Test";
        },
        loading() {
          return true;
        }
      }
    });

    // loading svg
    expect(
      wrapper
        .find(".sections__container--content > svg path > animatetransform")
        .exists()
    ).toBe(true);
    expect(wrapper.element).toMatchSnapshot();
  });
  it("Loading false, error true", () => {
    const wrapper = shallowMount(section_container, {
      computed: {
        title() {
          return "Title Test";
        },
        noDataMsg() {
          return "Warning Test";
        },
        error() {
          return "Error Test";
        },
        loading() {
          return false;
        }
      }
    });

    // loading svg
    expect(
      wrapper
        .find(".sections__container--content > svg path > animatetransform")
        .exists()
    ).toBe(false);
    expect(
      wrapper.find(".sections__container--content p.alert--danger").exists()
    ).toBe(true);
    expect(wrapper.element).toMatchSnapshot();
  });
  it("Loading false, error false", () => {
    const wrapper = shallowMount(section_container, {
      computed: {
        title() {
          return "Title Test";
        },
        noDataMsg() {
          return "Warning Test";
        },
        error() {
          return;
        },
        loading() {
          return false;
        }
      }
    });

    // loading svg
    expect(
      wrapper
        .find(".sections__container--content > svg path > animatetransform")
        .exists()
    ).toBe(false);
    expect(
      wrapper.find(".sections__container--content p.alert--warning").exists()
    ).toBe(true);
    expect(wrapper.element).toMatchSnapshot();
  });
});
