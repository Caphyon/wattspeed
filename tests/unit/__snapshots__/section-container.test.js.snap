// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Section-Container component Basic main container 1`] = `
<div
  class="sections__container __nodata"
>
  <div
    class="sections__container--title"
  >
    <svg
      class="section--icon"
      height="20"
      width="20"
    >
      <use
        xlink:href="assets/icons/symbols.svg#undefined"
        xmlns:xlink="http://www.w3.org/1999/xlink"
      />
    </svg>
     
    <h4
      class="mt0 mb0 text--strong"
    >
      Title Test
    </h4>
  </div>
   
  <div
    class="sections__container--content"
  >
    <svg
      height="40"
      viewBox="0 0 40 40"
      width="40"
      xmlns="http://www.w3.org/2000/svg"
      xmlns:xlink="http://www.w3.org/1999/xlink"
    >
      <path
        d="M20.201,5.169c-8.254,0-14.946,6.692-14.946,14.946c0,8.255,6.692,14.946,14.946,14.946 s14.946-6.691,14.946-14.946C35.146,11.861,28.455,5.169,20.201,5.169z M20.201,31.749c-6.425,0-11.634-5.208-11.634-11.634 c0-6.425,5.209-11.634,11.634-11.634c6.425,0,11.633,5.209,11.633,11.634C31.834,26.541,26.626,31.749,20.201,31.749z"
        opacity=".2"
      />
       
      <path
        d="M26.013,10.047l1.654-2.866c-2.198-1.272-4.743-2.012-7.466-2.012h0v3.312h0 C22.32,8.481,24.301,9.057,26.013,10.047z"
      >
        <animatetransform
          attributeName="transform"
          attributeType="xml"
          dur="0.7s"
          from="0 20 20"
          repeatCount="indefinite"
          to="360 20 20"
          type="rotate"
        />
      </path>
    </svg>
  </div>
</div>
`;

exports[`Testing Section-Container component Loading false, error false 1`] = `
<div
  class="sections__container"
>
  <div
    class="sections__container--title"
  >
    <svg
      class="section--icon"
      height="20"
      width="20"
    >
      <use
        xlink:href="assets/icons/symbols.svg#undefined"
        xmlns:xlink="http://www.w3.org/1999/xlink"
      />
    </svg>
     
    <h4
      class="mt0 mb0 text--strong"
    >
      Title Test
    </h4>
  </div>
   
  <div
    class="sections__container--content"
  >
    <p
      class="alert--warning"
    >
      
              Warning Test
            
    </p>
  </div>
</div>
`;

exports[`Testing Section-Container component Loading false, error true 1`] = `
<div
  class="sections__container"
>
  <div
    class="sections__container--title"
  >
    <svg
      class="section--icon"
      height="20"
      width="20"
    >
      <use
        xlink:href="assets/icons/symbols.svg#undefined"
        xmlns:xlink="http://www.w3.org/1999/xlink"
      />
    </svg>
     
    <h4
      class="mt0 mb0 text--strong"
    >
      Title Test
    </h4>
  </div>
   
  <div
    class="sections__container--content"
  >
    <p
      class="alert--danger"
    >
      
            ERROR: Error Test
          
    </p>
  </div>
</div>
`;

exports[`Testing Section-Container component Loading true 1`] = `
<div
  class="sections__container __nodata"
>
  <div
    class="sections__container--title"
  >
    <svg
      class="section--icon"
      height="20"
      width="20"
    >
      <use
        xlink:href="assets/icons/symbols.svg#undefined"
        xmlns:xlink="http://www.w3.org/1999/xlink"
      />
    </svg>
     
    <h4
      class="mt0 mb0 text--strong"
    >
      Title Test
    </h4>
  </div>
   
  <div
    class="sections__container--content"
  >
    <svg
      height="40"
      viewBox="0 0 40 40"
      width="40"
      xmlns="http://www.w3.org/2000/svg"
      xmlns:xlink="http://www.w3.org/1999/xlink"
    >
      <path
        d="M20.201,5.169c-8.254,0-14.946,6.692-14.946,14.946c0,8.255,6.692,14.946,14.946,14.946 s14.946-6.691,14.946-14.946C35.146,11.861,28.455,5.169,20.201,5.169z M20.201,31.749c-6.425,0-11.634-5.208-11.634-11.634 c0-6.425,5.209-11.634,11.634-11.634c6.425,0,11.633,5.209,11.633,11.634C31.834,26.541,26.626,31.749,20.201,31.749z"
        opacity=".2"
      />
       
      <path
        d="M26.013,10.047l1.654-2.866c-2.198-1.272-4.743-2.012-7.466-2.012h0v3.312h0 C22.32,8.481,24.301,9.057,26.013,10.047z"
      >
        <animatetransform
          attributeName="transform"
          attributeType="xml"
          dur="0.7s"
          from="0 20 20"
          repeatCount="indefinite"
          to="360 20 20"
          type="rotate"
        />
      </path>
    </svg>
  </div>
</div>
`;
